docs/topics/db/transactions.txt

```diff
@@ -176,6 +176,26 @@ Django provides a single API to control database transactions.
         If you catch exceptions raised by raw SQL queries, Django's behavior
         is unspecified and database-dependent.
 
+    .. note::
+
+        The model state will not be reverted whenever a transaction rollback
+        happens. This will lead to an inconsistent model state unless manually
+        reverted::
+
+            from django.db import DatabaseError, transaction
+
+            def update_object(object_id):
+                purchase = Purchase.objects.get(id=object_id)
```

Comment:
This doesn't demonstrate the sort of "manually reverting" that I had in mind. I envisioned something like this: 
```
orig_active = item.active
item.active = True
try:
    with transaction.atomic():
        item.save()
except DatabaseError:
    item.status = orig_active

if item.active:
    ...
```